backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/impl/ShipmentScheduleBL.java

```diff
@@ -968,4 +974,9 @@ public void setAsyncBatch(@NonNull final ShipmentScheduleId shipmentScheduleId,
 
 		shipmentSchedulePA.save(shipmentSchedule);
 	}
+
+	private boolean isMatchUsingOrderId()
+	{
+		return sysConfigBL.getBooleanValue(SYS_CONFIG_MATCH_USING_ORDER_ID, false);
```

Comment:
IMHO It's not OK to access a config of another module from here.
I would introduce something like IDesadvBL.isMatchUsingC_Order_ID().

In ShipmentScheduleBL I would reverse the dependency (because we cannot access IDesadvBL from here),
and I would 
* introduce a new interface called ShipmentScheduleAllowConsolidatePredicate, a functional interface with 1 method, `boolean isSchedAllowsConsolidate(final I_M_ShipmentSchedule sched)`
* introduce a new BL method IShipmentScheduleBL.addAllowConsolidatePredicate(ShipmentScheduleAllowConsolidatePredicate)
* change de.metas.inoutcandidate.api.impl.ShipmentScheduleBL#isSchedAllowsConsolidate and check the registered predicates

or do u have a better idea?

Reply:
@teosarca  @dragospodariu96 I don't see a better way either.
Also it's something we already use, so it's not a total WTF-moment when someone looks at he code in future.
(we just don't call i predicate so far - so i guess we should call the new interface also something with "..Vetoer"?)
See
- `de.metas.inoutcandidate.spi.ModelWithoutShipmentScheduleVetoer`
- `de.metas.shipment.ShipmentDeclarationVetoer`